src/data/lovelace_custom_cards.ts

```diff
@@ -0,0 +1,21 @@
+export interface CustomCardEntry {
+  type: string;
+  name?: string;
+  description?: string;
+  preview?: boolean;
+}
+
+export interface CustomCardsWindow {
+  customCards?: CustomCardEntry[];
+}
+
+const customCardsWindow = window as CustomCardsWindow;
+
+if (!("customCards" in customCardsWindow)) {
+  customCardsWindow.customCards = [];
+}
+
+export const customCards = customCardsWindow.customCards!;
+
+export const getCustomCardEntry = (type: string) =>
+  customCards.find((card) => `custom:${card.type}` === type);
```

Comment:
I still don't 100% agree that we should pass a type with `custom:` to here, but also don't care enough to block on it. 

The `custom:` prefix is a LL configuration implementation. If you look at just the custom card domain, there is no notion of `custom:`. If there was, one would expect custom cards to register themselves with `custom:` added to their type?

This is faster btw:

```suggestion
export const getCustomCardEntry = (type: string) => {
  const searchType = type.substr(7);
  return customCards.find((card) => card.type === searchType);
}
```

Reply:
The `custom:` prefix is required in several other places later, like the `getStubConfig`and `_createCardElement`. But I managed to make it far less magic at least...